fix: Configurable essentials max fix conflict of PR 8441

MiraiDevv opened the pull request:
resolve conflict of PR: #8441

blakegearin committed:
feat: Add config to support more than 12 essential tabs

blakegearin committed:
Merge branch 'dev' into configurable-essentials-max

Signed-off-by: Blake Gearin <hello@blakeg.me>

blakegearin committed:
Add dynamic getter

blakegearin committed:
Add overflow handling

mr-cheffy committed:
Merge branch 'dev' into configurable-essentials-max

blakegearin committed:
Merge branch 'dev' into configurable-essentials-max

Signed-off-by: Blake Gearin <hello@blakeg.me>

blakegearin committed:
Merge branch 'dev' into configurable-essentials-max

blakegearin committed:
Merge branch 'dev' into configurable-essentials-max

Signed-off-by: Blake Gearin <hello@blakeg.me>

blakegearin committed:
Fix merge conflict

Signed-off-by: Blake Gearin <hello@blakeg.me>

MiraiDevv committed:
Merge branch 'dev' into configurable-essentials-max

mr-cheffy commented:
Should we close #8441 ?

MiraiDevv commented:
> Should we close #8441 ?

I think so, I think the author hasn't seen it yet and it will probably take a while to see it, I decided to go ahead a little to see if you would accept the PR, but the author of this PR that I marked is not me, we are different people, just to be clear.

MiraiDevv committed:
Merge branch 'dev' into configurable-essentials-max

MiraiDevv commented:
Before merging this PR, I had one thought, what do you think about us establishing certain confirmation messages (pop-ups) to be opened, when certain amounts of essential tabs are changed? so for example, 4 essential tabs open => generate a warning that if you have 8gb of ram or less this may affect performance, 8 essential tabs open => generate a warning that if you have 16gb of ram or less this may affect browser performance and consume a lot of RAM (and this could considerably reduce most of the issues that people end up opening here on github, because if they read it, they will have an idea of ​​what is happening, because the warning was displayed), remembering that these warnings about the number of tabs and amount of ram are merely hypothetical, no calculation was made about this, this would just be a way to avoid many issues being opened for a problem that would be easily solved just by removing some essential tabs, anyway, please give your opinion on this and what you think about it, if this has already been discussed or not in the previous PR, I did see something like this being discussed, but not together with this new feature.

MiraiDevv committed:
Merge branch 'dev' into configurable-essentials-max

blakegearin commented on src/zen/workspaces/ZenWorkspaces.mjs:
```diff
@@ -2254,56 +2302,30 @@ var gZenWorkspaces = new (class extends ZenMultiWindowFeature {
-    name = 'Space',
-    icon = undefined,
-    dontChange = false,
+    name,
```
I don't recognize the changes from here down

MiraiDevv committed:
Merge branch 'dev' into configurable-essentials-max

MiraiDevv committed:
fix: put an initial value in the space name

The name value probably needs to have an initial value, so I put the default value back, but the icon, from what I understand from Zen and also from JavaScript, does not need to have an undefined value, since by default, without receiving any value, it is already undefined.

Signed-off-by: Mirai <mirai.dev@outlook.com>

MiraiDevv committed:
fix: Enhance pinned tabs resizing logic and workspace creation, including essential container updates and improved handling of extra tabs, b=no-bug, c=workspaces

MiraiDevv committed:
Merge branch 'configurable-essentials-max' of https://github.com/MiraiDevv/desktop into configurable-essentials-max

MiraiDevv commented:
I realized that I had removed some things that were important, I put them back looking at the code that was there before, if anyone has any more changes that need to be made or thinks that something is different, you can let me know, please, have a good day.

MiraiDevv committed:
Merge branch 'dev' into configurable-essentials-max

mr-cheffy commented:
This should be without a second pref for height, it should be dynamic

MiraiDevv committed:
refactor: Remove essentials height preference and related logic from ZenUIManager and CSS, simplifying the vertical tabs implementation

MiraiDevv committed:
Merge branch 'configurable-essentials-max' of https://github.com/MiraiDevv/desktop into configurable-essentials-max

MiraiDevv commented:
> This should be without a second pref for height, it should be dynamic

Is that what you meant? From what I understood, it was just to remove the logic that was determining the height, if not, if you can point to the line/file or explain more comprehensively, if it's not too much trouble.

MiraiDevv committed:
refactor: Remove the onPinnedTabsResize method to speed up workspace tab management (it had another identical method above).

mr-cheffy commented:
That's exactly what I mean, why is [ZenWorkspaces.mjs](https://github.com/zen-browser/desktop/pull/9189/files#diff-1cb3ec312afba298664bd57b1ea99e8a710e4a92ebb378ab39984ac9e0e482c0) a new created file though?

MiraiDevv commented:
> That's exactly what I mean, why is [ZenWorkspaces.mjs](https://github.com/zen-browser/desktop/pull/9189/files#diff-1cb3ec312afba298664bd57b1ea99e8a710e4a92ebb378ab39984ac9e0e482c0) a new created file though?

I didn't understand either, I just remember that I accidentally removed some lines and then put them back, the last time I compared the two files they were the same, I won't be sure just by saying, but I think it was a github bug, because I didn't even delete the file...........

blakegearin commented:
Just wanted to call out this has been awaiting approval for over 60 days now (factoring in both PRs)

MiraiDevv commented:
> Just wanted to call out this has been awaiting approval for over 60 days now (factoring in both PRs)

I also thought about saying this, I think cheff could put other people to review the PR and decouple the work........but that's just a personal opinion.

blakegearin commented:
@MiraiDevv This needs rebasing again. We've reached the 100 day mark. Starting to wonder if Zen is sustainable if contributions aren't welcomed. The code you were asked to remove is the same code I was asked to add back in May

MiraiDevv commented:
> @MiraiDevv This needs rebasing again. We've reached the 100 day mark. Starting to wonder if Zen is sustainable if contributions aren't welcomed. The code you were asked to remove is the same code I was asked to add back in May

@blakegearin I'm wondering the same thing, even though I really have a lot of PRs, it would be impossible not to have had time to at least delegate this to some contributor that Mauro probably trusted "more" to merge the changes, I think the best thing would be to leave both PRs closed and create a new one or simply leave it and do this in another browser, since initially Mauro was reluctant about the idea of ​​giving a configuration regarding the essential tabs.............I don't know if that was the reason he didn't respond more or talk about it, but, right.......

MiraiDevv commented:
> > @MiraiDevv This needs rebasing again. We've reached the 100 day mark. Starting to wonder if Zen is sustainable if contributions aren't welcomed. The code you were asked to remove is the same code I was asked to add back in May
> 
> @blakegearin I'm wondering the same thing, even though I really have a lot of PRs, it would be impossible not to have had time to at least delegate this to some contributor that Mauro probably trusted "more" to merge the changes, I think the best thing would be to leave both PRs closed and create a new one or simply leave it and do this in another browser, since initially Mauro was reluctant about the idea of ​​giving a configuration regarding the essential tabs.............I don't know if that was the reason he didn't respond more or talk about it, but, right.......

Anyway, I bought a new computer and it will arrive before or by the 4th, with that I will be able to build Zen from the source code. If you want, I can close this PR and from the 4th I can take what was done on yours and here and create a new PR, but Mauro probably won't approve the changes......

mr-cheffy commented on src/zen/workspaces/ZenWorkspaces.mjs:
```diff
@@ -311,6 +311,70 @@ var gZenWorkspaces = new (class extends ZenMultiWindowFeature {
     }
   }
 
+  async onPinnedTabsResize(entries, forAnimation = false, animateContainer = false) {
```
Why was this added?

mr-cheffy commented on src/zen/workspaces/ZenWorkspaces.mjs:
```diff
@@ -2264,11 +2330,11 @@ var gZenWorkspaces = new (class extends ZenMultiWindowFeature {
   }
 
   async createAndSaveWorkspace(
-    name = 'Space',
```
Same here, why was a change done here

mr-cheffy commented on ZenWorkspaces.mjs:
```diff
@@ -0,0 +1,2928 @@
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+var gZenWorkspaces = new (class extends ZenMultiWindowFeature {
```
Same here

mr-cheffy commented on src/zen/tabs/zen-tabs/vertical-tabs.css:
```diff
@@ -1322,6 +1326,48 @@
+
+    &::after {
+      content: '';
+      position: sticky;
```
Why sticky possition?

mr-cheffy reviewed (changes requested):
@MiraiDevv please request reviews from me because otherwise it will be lost

FujiwaraChoki commented:
Thank you for working hard @mr-cheffy

blakegearin commented:
@MiraiDevv Do you plan on updating this branch?

blakegearin commented:
I will be starting a fresh branch if @MiraiDevv doesn't respond soon.

mr-cheffy commented:
I'll close this then
